auto_comps.py

```diff
@@ -0,0 +1,224 @@
+    deck_soup = BeautifulSoup(input_str, "lxml")
+    deck_list = []
+    for deck in deck_soup.select(".guide-meta__deck-box"):
+        nms = deck.select_one(".guide-meta__deck__column.name.mr-3").find(text=True).strip()
+        afs = deck.select_one(".open-builder>a").attrs['href']
+        deck_list.append((nms, afs))
+    for nms, afs in deck_list:
+        deck_keys = parse.parse_qs(parse.urlparse(afs).query)['deck'][0]
+        deck_response = requests.get("https://lolchess.gg/builder/set8?hl=en&deck=" + deck_keys)
+        pattern = r'<script id="__NEXT_DATA__" type="application/json">\s*({[\s\S]*?})\s*</script>'
+        json_in_text = re.search(pattern, deck_response.text).group(1)
+        query_data = json.loads(json_in_text).get("props").get("pageProps").get("dehydratedState").get("queries")[
+            1].get("state").get("data").get("refs")
+        with open("cached_data/deck.json", "w") as f:
+            f.write(json.dumps(query_data))
+        deck_slots = requests.get(f"https://tft.dakgg.io/api/v1/team-builders/{deck_keys}").json()
+        slots = {}
+        counter = 0
+        augments = deck_slots.get("teamBuilder", {}).get("augments", [])
+        real_augments = [arg.get("name") for arg in query_data.get(""
+                                                                   "", []) if arg.get("key") in augments]
+        for each_slot in deck_slots.get("teamBuilder", {}).get("slots", []):
+
+            if each_slot is not None:
+                try:
+                    champion_name = \
+                    list(filter(lambda e: e['key'] == each_slot.get("champion"), query_data.get("champions")))[
+                        0]['name']
+                except:
+                    continue
+                slot_items = render_item(query_data.get("items"), each_slot.get("items", []))
+                slots[champion_name] = {'board_position': LOLCHESS_BOARD_ARRANGE[each_slot.get("index")], 'items': slot_items,
+                                        'level': 3, 'final_comp': True}
+            counter += 3
+
+        output_comps.append([nms, slots, real_augments])
+    return output_comps
+
+
+def render_item(ob, ids):
+    items = list(filter(lambda e: e['key'] in ids, ob))
+    return [i['name'].replace(" ", "").replace("'", "") for i in items]
+
+
+def __LoadCommunityDragon():
+    loaded_string = requests.get(DRAGON_URL)
+    json_parsed = json.loads(loaded_string.text)
+    set_data_champions = json_parsed["setData"][2]["champions"]
+    output = {}
+    for each_champion in set_data_champions:
+        board_size = 1
+        traits = each_champion["traits"]
+        for each_trait in traits:
+            if each_trait == 'Dragon':
+                board_size = 2
+                break
+        name_tft = each_champion["apiName"]
+        ingame_name = each_champion["name"]
+        if name_tft not in output:
+            output[name_tft] = (ingame_name, board_size)
+    return output
```

Comment:
Simplifying and refactoring some of the logic here:
```suggestion
    for each_champion in set_data_champions:
        traits = each_champion["traits"]
        board_size = next((2 for each_trait in traits if each_trait == 'Dragon'), 1)
        name_tft = each_champion["apiName"]
        if name_tft not in output:
            ingame_name = each_champion["name"]
            output[name_tft] = (ingame_name, board_size)
    return output
```